src/Illuminate/Support/Manager.php

```diff
@@ -31,12 +39,13 @@ abstract class Manager
     /**
      * Create a new manager instance.
      *
-     * @param  \Illuminate\Contracts\Container\Container|\Illuminate\Contracts\Foundation\Application  $app
+     * @param  \Illuminate\Contracts\Container\Container  $container
      * @return void
      */
-    public function __construct($app)
+    public function __construct(Container $container)
     {
-        $this->app = $app;
+        $this->config = $container->make('config');
+        $this->container = $container;
```

Comment:
Renaming `$this->app` to `$this->container` would mean any packages that extends `Illuminate\Support\Manager` would have to split version just for 6.0 releases (instead of just updating `composer.json` if everything else is backward compatible with 5.8 and below).

Reply:
I'm against making a BC here because other `Manager` that doesn't appends `Illuminate\Support\Manager` still uses `$this->app` as property name.

Reply:
Hmm yeah you're right. We'll have to do a new major release of Socialite for example for this. 

But does keeping an incorrect property name around make sense? When would be a good time to change this then?

Reply:
@crynobone I'll send in a PR to revert that change.

Reply:
> I'm against making a BC here because other Manager that doesn't appends Illuminate\Support\Manager still uses $this->app as property name.

All of these examples require the Application itself because they make specific calls on it that are only available on the Application instance.

Reply:
> All of these examples require the Application itself because they make specific calls on it that are only available on the Application instance.

Isn't `DatabaseManager` being used in Lumen?

Reply:
Yeah, and that leads us again to the discussion about the Application contract :-/